File: sources/CSharp/src/Ers/SubModel/Event/SubModelSignals.cs
using System;
using System.Runtime.InteropServices;

#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
namespace Ers
{
    [Obsolete("Replaced by overridable methods (OnEntering, OnEntered, OnExiting, OnExited) on ScriptBehavior")]
    [StructLayout(LayoutKind.Sequential)]
    public ref struct SubModelSignals
    {
        public RelationEvents Relation;

        public SubModelSignals(IntPtr subModel) { Relation = new RelationEvents(subModel); }
    }
}
#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member

File: sources/CPP/include/Ers/SubModel/Event/SubModelSignals.cpp
#include "SubModelSignals.h"

void* Ers::SubModelSignals::Data()
{
    return this;
}

const void* const Ers::SubModelSignals::Data() const
{
    return this;
}

Ers::RelationEvents& Ers::SubModelSignals::Relation()
{
    return *static_cast<RelationEvents*>(Data());
}
